osm2lanes/src/algorithm.rs

```diff
@@ -1,610 +1,238 @@
-    } else {
-        LaneType::Driving
-    };
-
-    // These are ordered from the road center, going outwards. Most of the members of fwd_side will
-    // have Direction::Fwd, but there can be exceptions with two-way cycletracks.
-    let mut fwd_side: Vec<LaneSpec> = iter::repeat_with(|| fwd(driving_lane))
-        .take(num_driving_fwd)
-        .collect();
-    let back_side: Vec<LaneSpec> = iter::repeat_with(|| back(driving_lane))
-        .take(num_driving_back)
-        .collect();
-    if tags.is("lanes:both_ways", "1") || tags.is("centre_turn_lane", "yes") {
-        fwd_side.insert(0, fwd(LaneType::SharedLeftTurn));
+        true
     }
-
-    (fwd_side, back_side, oneway, driving_lane)
 }
 
-fn add_bus_lanes(
-    fwd_side: &mut Vec<LaneSpec>,
-    back_side: &mut Vec<LaneSpec>,
-    oneway: bool,
-    tags: &Tags,
-    cfg: &MapConfig,
-) {
-    let fwd_bus_spec = if let Some(s) = tags.get("bus:lanes:forward") {
-        s
-    } else if let Some(s) = tags.get("psv:lanes:forward") {
-        s
-    } else if oneway {
-        if let Some(s) = tags.get("bus:lanes") {
-            s
-        } else if let Some(s) = tags.get("psv:lanes") {
-            s
-        } else {
-            ""
-        }
-    } else {
-        ""
-    };
-    if !fwd_bus_spec.is_empty() {
-        let mut parts: Vec<&str> = fwd_bus_spec.split('|').collect();
-        let offset = if fwd_side[0].lt == LaneType::SharedLeftTurn {
-            1
-        } else {
-            0
-        };
-        // Per https://wiki.openstreetmap.org/wiki/Lanes#Description, the parts are ordered
-        // left-to-right when facing forwards. fwd_side is in-to-out, which is left-to-right only
-        // for right-handed driving. So for left-handed, invert.
-        if cfg.driving_side == DrivingSide::Left {
-            parts.reverse();
-        }
-        if parts.len() == fwd_side.len() - offset {
-            for (idx, part) in parts.into_iter().enumerate() {
-                if part == "designated" {
-                    fwd_side[idx + offset].lt = LaneType::Bus;
-                }
-            }
-        }
-    }
-    if let Some(spec) = tags
-        .get("bus:lanes:backward")
-        .or_else(|| tags.get("psv:lanes:backward"))
-    {
-        let mut parts: Vec<&str> = spec.split('|').collect();
-        // Again, the parts are ordered left-to-right when facing backwards. back_side is
-        // in-to-out, which is left-to-right only for right-handed driving. So for left-handed,
-        // invert.
-        if cfg.driving_side == DrivingSide::Left {
-            parts.reverse();
-        }
-        if parts.len() == back_side.len() {
-            for (idx, part) in parts.into_iter().enumerate() {
-                if part == "designated" {
-                    back_side[idx].lt = LaneType::Bus;
-                }
-            }
-        }
-    }
+fn access_level_allowed(access: AccessLevel) -> bool {
+    matches!(
+        access,
+        AccessLevel::Designated
```

Comment:
> What're your thoughts?

I'm not entirely sure about this function either. As far as I understood the old algorithm checked `access != No` so this might just be the easiest way to implement this function. Maybe we could add a case for `Private`.

I'm, however, not sure whether it makes more sense to be more permissive (allowing `Destination`, `Permit`, or even `Agricultural`) or whether being more restrictive (only `Yes`, `Designated` and `Permissive`) makes more sense considering the downstream users such as the A/B Street routing engine.